src/basekernels/nn.jl

```diff
@@ -20,7 +20,35 @@ Bayesian neural network with erf (Error Function) as activation function.
 struct NeuralNetworkKernel <: Kernel end
 
 function (κ::NeuralNetworkKernel)(x, y)
-    return asin(dot(x, y) / sqrt((1 + sum(abs2, x)) * (1 + sum(abs2, y))))
+    return asin(dot(x, y) / sqrt((1 + sum(abs2.(x))) * (1 + sum(abs2.(y)))))
+end
+
+function kernelmatrix(::NeuralNetworkKernel, x::ColVecs, y::ColVecs)
+    validate_inputs(x, y)
+    X_2 = sum(x.X .* x.X, dims=1)
+    Y_2 = sum(y.X .* y.X, dims=1)
+    XY = x.X' * y.X
+    return asin.(XY ./ sqrt.((X_2 .+ 1.0)' * (Y_2 .+ 1.0)))
```

Comment:
One should avoid literals of type Float64 since it will promote the computations (which is not intended, e.g., if you work with Float32).
```suggestion
    return asin.(XY ./ sqrt.((X_2 .+ 1)' * (Y_2 .+ 1)))
```